src/main/kotlin/app/CodeLongevity.kt

```diff
@@ -0,0 +1,231 @@
+            if (author.getEmailAddress() != email) {
+                continue
+            }
+            //println("got a line, age ${line.age}\n ${line.printme()}")
+            sum += line.age
+            total++
+        }
+
+        //println("All lines:")
+        //codeLines.forEach { line -> line.printme() }
+
+        var avg = if (total > 0) sum / total else 0
+        println("avg code line age for <$email> is ${avg} seconds")
+    }
+
+    /**
+     * Computes age of all lines in the repo through all revisions.
+     */
+    private fun compute() {
+        val treeWalk = TreeWalk(repo)
+        treeWalk.setRecursive(true)
+        treeWalk.addTree(head!!.getTree())
+
+        val files: MutableMap<String, ArrayList<RevCommitLine>> = mutableMapOf()
+
+        // Build a map of file names and their code lines.
+        while (treeWalk.next()) {
+            // TODO(alex): skip binary files.
+            val fileName = treeWalk.getPathString()
+            val fileLoader = repo.open(treeWalk.getObjectId(0))
+            val fileText = RawText(fileLoader.getBytes())
+            val linesNum = fileText.size()
+            var lines = ArrayList<RevCommitLine>(linesNum)
+            var idx = 0
+            while (idx < fileText.size()) {
+                lines.add(RevCommitLine(head, fileName, idx))
+                idx++
+            }
+            files.put(fileName, lines)
+        }
+  
+        val df = DiffFormatter(DisabledOutputStream.INSTANCE)
+        df.setRepository(repo)
+        df.setDetectRenames(true)
+
+        val revWalk = RevWalk(repo)
+        revWalk.markStart(head)
+
+        var commit: RevCommit? = head
+        var parentCommit: RevCommit? = revWalk.next()  // proceed to head
+        do {
+            parentCommit = revWalk.next()
+
+            println("commit: ${commit!!.getName()}; '${commit.getShortMessage()}'")
+            if (parentCommit != null) {
+                println("parent commit: ${parentCommit.getName()}; '${parentCommit.getShortMessage()}'")
+            }
+            else {
+                println("parent commit: null")
+            }
+
+            val diffs = df.scan(parentCommit, commit)
+
+            for (diff in diffs) {
+                // XXX(alex): does it happen in the wilds?
+                if (diff.changeType == DiffEntry.ChangeType.COPY) {
+                    continue
+                }
+
+                // File was renamed, tweak the files map.
+                if (diff.changeType == DiffEntry.ChangeType.RENAME) {
+                    println("old path: ${diff.getOldPath()}")
+                    println("new path: ${diff.getNewPath()}")
+                    files.set(diff.getOldPath(),
+                              files.remove(diff.getNewPath())!!)
+                    println("oopsy")
+                    //continue
+                }
+
+                // Get a edit list and traverse it backwards to avoid indices
+                // adjustment.
```

Comment:
I have a bunch of ins/del chunks with indices relative original/new file, so I have to traverse the chunks array from back to use ins/del indices unchanged to ins/del to/from the lines array. Otherwise ins/del indices won't much indices of the lines array.